### PR TITLE
Bug 1865904 - Add correct default elevation to FAB

### PR DESCRIPTION
**Pull request by MozillaNoah**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1865904

### CODE REVIEW

File: fenix/app/src/main/java/org/mozilla/fenix/compose/button/FloatingActionButton.kt [hunk 1]
```diff
@@ -35,6 +37,7 @@ import org.mozilla.fenix.theme.FirefoxTheme
  * @param modifier [Modifier] to be applied to the action button.
  * @param contentDescription The content description to describe the icon.
  * @param label Text to be displayed next to the icon.
+ * @param elevation Elevation parameter set.
```
Comment (gabrielluong) [inline fenix/app/src/main/java/org/mozilla/fenix/compose/button/FloatingActionButton.kt:40]:
```suggestion
 * @param elevation  FloatingActionButtonElevation used to resolve the elevation for this FAB in different states.
```
This is from the `FloatingActionButton` kDoc from for this parameter.

File: fenix/app/src/main/java/org/mozilla/fenix/compose/button/FloatingActionButton.kt [hunk 2]
```diff
@@ -43,13 +46,15 @@ fun FloatingActionButton(
     modifier: Modifier = Modifier,
     contentDescription: String? = null,
     label: String? = null,
+    elevation: FloatingActionButtonElevation = FloatingActionButtonDefaults.elevation(defaultElevation = 5.dp),
```
Comment (gabrielluong) [inline fenix/app/src/main/java/org/mozilla/fenix/compose/button/FloatingActionButton.kt:49]: I do see in the figma that we have 3 values 5, 18, 10 dp which probably represents the other states. I know we only care about the default state from the ticket, but is there any reason we are not adjusting the elevation for the hovered, pressed and focused state while we are already adjusting the elevation here?

I think the default values for the other state are fine, but unclear why we are excluding the elevation.

**Review by gabrielluong — APPROVED** [review]

File: fenix/app/src/main/java/org/mozilla/fenix/compose/button/FloatingActionButton.kt (hunk 2 above)
Comment (MozillaNoah) [reply fenix/app/src/main/java/org/mozilla/fenix/compose/button/FloatingActionButton.kt:49]: I'll be honest, I have no idea how Figma is deriving the 18 and 10 values in the Compose code generator tool. I got the single value of 5 by checking the XML code.

As far as I can tell, the design system doesn't have a language for these other states, so I just wanted to get this change landed and defer that conversation for now, as we have no precedence for specifically setting the elevation (or colors) for pressed and hovered states.

Comment (gabrielluong) [reply fenix/app/src/main/java/org/mozilla/fenix/compose/button/FloatingActionButton.kt:49]: sgtm

**Pull request merged**